.github/workflows/build-api-ref.yml

```diff
@@ -0,0 +1,60 @@
+name: Generate API Documentation
+
+on:
+  push:
+    branches:
+      - 'main'   # This will only trigger the workflow on the main branch
+    paths:
+      - 'res/**' # This will only trigger on changes to the /res/ folder
+  workflow_dispatch:
+
+jobs:
+  generate-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'
+
+    - name: Install dependencies
+      run: |
+        pip install pyyaml
+        pip install pytest
+
+    - name: Run tests
+      working-directory: redocly
+      run: pytest ./scripts/combine_yamls_test.py
+
+    - name: Run Python script to combine YAMLs
+      working-directory: redocly  # Run script in the redocly folder
+      run: |
+        mkdir -p output
+        python ./scripts/combine_yamls.py
```

Comment:
Comment for future: the script should take parameters with input and output files so we don't have filenames in both workflows and and utility files. Better to just define them in the workflow.